data_model/src/trigger.rs

```diff
@@ -95,50 +206,226 @@ impl Action {
+        Action::<FilterBox>::new(
+            self.executable.clone(),
+            self.repeats.clone(),
+            self.technical_account.clone(),
+            self.filter.clone().into(),
+        )
     }
 }
 
-impl Ord for Action {
-    fn cmp(&self, other: &Self) -> Ordering {
+impl<F: Filter + PartialEq> PartialOrd for Action<F> {
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
         // Exclude the executable. When debugging and replacing
         // the trigger, its position in Hash and Tree maps should
         // not change depending on the content.
         match self.repeats.cmp(&other.repeats) {
-            Ordering::Equal => {}
-            ord => return ord,
+            cmp::Ordering::Equal => {}
+            ord => return Some(ord),
         }
-        self.technical_account.cmp(&other.technical_account)
+        Some(self.technical_account.cmp(&other.technical_account))
+    }
+}
+
+#[allow(clippy::expect_used)]
+impl<F: Filter + Eq> Ord for Action<F> {
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        self.partial_cmp(other)
+            .expect("`PartialCmp::partial_cmp()` for `Action` should never return `None`")
     }
 }
 
 /// Enumeration of possible repetitions schemes.
-#[derive(
-    Debug,
-    Clone,
-    Copy,
-    PartialOrd,
-    Ord,
-    PartialEq,
-    Eq,
-    Encode,
-    Decode,
-    Serialize,
-    Deserialize,
-    IntoSchema,
-)]
+#[derive(Debug, Clone, Encode, Decode, Serialize, Deserialize, IntoSchema)]
 pub enum Repeats {
     /// Repeat indefinitely, until the trigger is unregistered.
     Indefinitely,
     /// Repeat a set number of times
-    Exactly(u32), // If you need more, use `Indefinitely`.
+    Exactly(AtomicU32Wrapper), // If you need more, use `Indefinitely`.
+}
+
+impl PartialOrd for Repeats {
+    fn partial_cmp(&self, other: &Self) -> Option<cmp::Ordering> {
+        Some(self.cmp(other))
+    }
 }
 
+impl Ord for Repeats {
+    fn cmp(&self, other: &Self) -> cmp::Ordering {
+        match (self, other) {
+            (Repeats::Indefinitely, Repeats::Indefinitely) => cmp::Ordering::Equal,
+            (Repeats::Indefinitely, Repeats::Exactly(_)) => cmp::Ordering::Greater,
+            (Repeats::Exactly(_), Repeats::Indefinitely) => cmp::Ordering::Less,
+            (Repeats::Exactly(l), Repeats::Exactly(r)) => l.cmp(r),
+        }
+    }
+}
+
+impl PartialEq for Repeats {
+    fn eq(&self, other: &Self) -> bool {
+        match (self, other) {
+            (Self::Exactly(l0), Self::Exactly(r0)) => l0 == r0,
+            _ => false,
+        }
+    }
+}
+
+impl Eq for Repeats {}
+
 impl From<u32> for Repeats {
     fn from(num: u32) -> Self {
-        Repeats::Exactly(num)
+        Repeats::Exactly(AtomicU32Wrapper::new(num))
+    }
+}
+
+/// Wrapper for [`AtomicU32`]
+///
+/// Provides useful impls, using [`Ordering::Acquire`]
+/// and [`Ordering::Release`] to load and store respectively
+#[derive(Debug)]
+pub struct AtomicU32Wrapper(AtomicU32);
```

Comment:
1. Please move this to a separate module in `primitives`, because it was designed to hold simple data structure wrappers. 
2. I don't think it needs to be called `AtomicU32Wrapper`, just `AtomicU32`. Instead of `use core::sync::atomic`, it'll be imported from `primitives` and we'll be fine. 
3. All of the `impls` **must** be marked as `inline` or they will have a huge performance overhead without LTO.